pkg/operator/k8sutil/resources.go

```diff
@@ -85,11 +86,32 @@ func (info *OwnerInfo) SetOwnerReference(object metav1.Object) error {
 	if err != nil {
 		return err
 	}
-	ownerRefs := append(object.GetOwnerReferences(), *info.ownerRef)
+	ownerRefs := object.GetOwnerReferences()
+	for _, v := range ownerRefs {
+		if referSameObject(v, *info.ownerRef) {
+			return nil
+		}
+	}
+	ownerRefs = append(ownerRefs, *info.ownerRef)
 	object.SetOwnerReferences(ownerRefs)
 	return nil
 }
 
+//This logic is copied from https://github.com/kubernetes-sigs/controller-runtime/blob/a905949b9040084f0c6d2a27ec70e77c3c5c0931/pkg/controller/controllerutil/controllerutil.go#L160
+func referSameObject(a, b metav1.OwnerReference) bool {
+	aGV, err := schema.ParseGroupVersion(a.APIVersion)
+	if err != nil {
+		return false
+	}
+
+	bGV, err := schema.ParseGroupVersion(b.APIVersion)
+	if err != nil {
+		return false
+	}
+
+	return aGV.Group == bGV.Group && a.Kind == b.Kind && a.Name == b.Name
```

Comment:
What are examples of the group versions? Could we just do a string compare on `a.APIVersion == b.APIVersion`, or why do we need to fully parse all the fields?

Reply:
@travisn Let assume the group version of an ownerRef is "example.com/v1beta". If the api version of the corresponding object is updated to "v1", we shouldn't distinguish two group versions[1]. Otherwise, ownerRef duplication would happen.

[1] https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups-and-versioning
> API resources are distinguished by their API group, resource type, namespace (for namespaced resources), and name.